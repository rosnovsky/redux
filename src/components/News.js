import React, { useState, useEffect } from 'react';
import store from '../store';
import '../styles/App.css';

// as soon as an headline is selected in the Headline component, we will take the article from the store and display it. We will have MARK_AS_READ action (removes article from the page, and removes its headline) and READ_LATER (tucks an article away in ReadLater component).

function News({ article }) {
    const [activeArticle, setActiveArticle] = useState();

    const updateArticle = () => {
        setActiveArticle(store.getState().activeArticle);
    };

    store.subscribe(updateArticle);

    return (
        <div className="App">
            <h2>News</h2>
            <h3>{article.title}</h3>
            <p>{article.content}</p>
        </div>
    );
}

export default News;
