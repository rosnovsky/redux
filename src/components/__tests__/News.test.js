import React from 'react';
import renderer from 'react-test-renderer';
import News from '../News';
import { Provider } from 'react-redux';
import store from '../../store';

describe('News', () => {
    it('News component renders without crashing', () => {
        const app = renderer.create(<News />);
        let tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
