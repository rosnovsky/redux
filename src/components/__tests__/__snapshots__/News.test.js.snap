// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News News component renders without crashing 1`] = `
<div
  className="App"
>
  <h2>
    News
  </h2>
</div>
`;
